pkg/delete/delete_test.go

```diff
@@ -1149,93 +1148,48 @@ func TestDelete_Documents(t *testing.T) {
 	})
 }
 
-type segmentStubClient struct {
-	called bool
-	list   func() (segments.Response, error)
-	getAll func() ([]segments.Response, error)
-	delete func() (segments.Response, error)
-}
-
-func (c *segmentStubClient) List(_ context.Context) (segments.Response, error) {
-	return c.list()
-}
-
-func (c *segmentStubClient) GetAll(_ context.Context) ([]segments.Response, error) {
-	return c.getAll()
-}
-
-func (c *segmentStubClient) Delete(_ context.Context, _ string) (segments.Response, error) {
-	c.called = true
-	return c.delete()
-}
-
 func TestDelete_Segments(t *testing.T) {
-	t.Run("simple case", func(t *testing.T) {
-		t.Setenv(featureflags.Segments.EnvName(), "true")
-
-		c := segmentStubClient{
-			delete: func() (segments.Response, error) {
-				return segments.Response{StatusCode: http.StatusOK}, nil
+	c := client.TestSegmentsClient{}
+	given := delete.DeleteEntries{
+		"segment": {
+			{
+				Type:           "segment",
+				OriginObjectId: "originObjectID",
 			},
-		}
+		},
+	}
+
+	t.Run("With Enabled Segment FF", func(t *testing.T) {
+		t.Setenv(featureflags.Segments.EnvName(), "true")
 
-		given := delete.DeleteEntries{
-			"segment": {
-				{
-					Type:           "segment",
-					OriginObjectId: "originObjectID",
-				},
-			},
-		}
 		err := delete.Configs(context.TODO(), client.ClientSet{SegmentClient: &c}, given)
-		assert.NoError(t, err)
-		assert.True(t, c.called, "delete should have been called")
+		//DummyClient returns unimplemented error on every execution of any method
```

Comment:
I think we need to update this too:
```suggestion
		// fakeClient returns unimplemented error on every execution of any method
```